src/components/course/EnrollButton.jsx

```diff
@@ -1,67 +1,45 @@
-import React, { useContext, useMemo, useState } from 'react';
-import PropTypes from 'prop-types';
-import classNames from 'classnames';
-import { Link, useLocation } from 'react-router-dom';
+import React, { useContext, useMemo } from 'react';
+import { useLocation } from 'react-router-dom';
 import { AppContext } from '@edx/frontend-platform/react';
-import { Button } from '@edx/paragon';
 
-import { UserSubsidyContext } from '../enterprise-user-subsidy';
 import { CourseContext } from './CourseContextProvider';
 import EnrollButtonLabel from './EnrollButtonLabel';
-import EnrollModal from './EnrollModal';
-
-import { features } from '../../config';
+import EnrollAction from './enrollment/EnrollAction';
 
+import { determineEnrollmentType } from './enrollment/utils';
 import { useCourseEnrollmentUrl } from './data/hooks';
 
-import {
-  hasCourseStarted,
-  findUserEnrollmentForCourseRun,
-  findHighestLevelSeatSku,
-  findOfferForCourse,
-  shouldUpgradeUserEnrollment,
-} from './data/utils';
-
-const EnrollButtonWrapper = ({
-  as: Component,
-  children,
-  ...props
-}) => (
-  <div className="enroll-wrapper" style={{ width: 270 }}>
-    <Component {...props}>
-      {children}
-    </Component>
-  </div>
-);
-
-EnrollButtonWrapper.propTypes = {
-  children: PropTypes.node.isRequired,
-  as: PropTypes.elementType,
-};
-
-EnrollButtonWrapper.defaultProps = {
-  as: Button,
-};
+import { findHighestLevelSeatSku } from './data/utils';
+import { useSubsidyDataForCourse, useEnrollData } from './enrollment/hooks';
 
 export default function EnrollButton() {
   const { enterpriseConfig } = useContext(AppContext);
   const { state: courseData } = useContext(CourseContext);
-  const { subscriptionLicense, offers: { offers, offersCount } } = useContext(UserSubsidyContext);
+
+  const {
+    subscriptionLicense,
+    userSubsidyApplicableToCourse,
+    courseHasOffer,
+  } = useSubsidyDataForCourse();
   const location = useLocation();
-  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const {
+    isUserEnrolled,
+    isCourseStarted,
+    isEnrollable,
+    userEnrollment,
+  } = useEnrollData();
 
   const {
```

Comment:
this area of code to do enrollmenturl and enrolllabel will be further cleaned out once the label refactor is done (separate ticket)